website/src/views/pageStateHandlers/dateFilterFromToUrl.ts

```diff
@@ -0,0 +1,44 @@
+import type { DateRangeOption } from '@genspectrum/dashboard-components/util';
+
+import type { BaselineFilterConfig } from '../../components/pageStateSelectors/BaselineSelector.tsx';
+import type { Dataset } from '../View.ts';
+import { getDateRangeFromSearch, setSearchFromDateRange } from '../helpers.ts';
+
+export function parseDateRangesFromUrl(
+    search: URLSearchParams | Map<string, string>,
+    baselineFilterConfigs: BaselineFilterConfig[] | undefined,
+) {
+    const dateRangeFilterConfigs = baselineFilterConfigs?.filter((config) => config.type === 'date');
+
+    return (
+        dateRangeFilterConfigs?.reduce(
+            (acc, config) => {
+                const dateRange = getDateRangeFromSearch(search, config.dateColumn, config.dateRangeOptions);
```

Comment:
We could move `getDateRangeFromSearch` into this file, too.

Reply:
(Applies to a bunch of other functions in the helpers, too, but we can also do that later.)